packages/sanity/src/structure/structureBuilder/util/resolveTypeForDocument.ts

```diff
@@ -1,22 +1,15 @@
 import {type SanityClient} from '@sanity/client'
-import {
-  DEFAULT_STUDIO_CLIENT_OPTIONS,
-  getDraftId,
-  getPublishedId,
-  type SourceClientOptions,
-} from 'sanity'
+import {DEFAULT_STUDIO_CLIENT_OPTIONS, getPublishedId, type SourceClientOptions} from 'sanity'
 
 export async function resolveTypeForDocument(
   getClient: (options: SourceClientOptions) => SanityClient,
   id: string,
 ): Promise<string | undefined> {
-  const query = '*[_id in [$documentId, $draftId]]._type'
-  const documentId = getPublishedId(id)
-  const draftId = getDraftId(id)
+  const query = '*[sanity::versionOf($publishedId)]._type'
```

Comment:
```suggestion
  const query = '*[sanity::versionOf($publishedId)]._type[0]'
```
And then simplify by just:
```ts
return getClient(DEFAULT_STUDIO_CLIENT_OPTIONS).fetch(
    query,
    {publishedId: getPublishedId(id)},
    {tag: 'structure.resolve-type'},
  )
```